matrixcomponent/ontology.py

```diff
@@ -0,0 +1,232 @@
+from typing import List
+from rdflib import Namespace, Graph, Literal, URIRef, RDF
+
+
+class Region:
+    ns: URIRef
+    begin: int
+    end: int
+
+    def __str__(self):
+        suffix = str(self.begin)  # region/2
+        if self.begin != self.end:
+            suffix = suffix + "-" + str(self.end)  # region/2-10
+        return "region/" + suffix
+
+    def ns_term(self):
+        return self.ns + str(self) # path1/...
+
+    def add_to_graph(self, graph: Graph, vg, faldo: Namespace) -> None:
+        region = self.ns_term()  # str representation
+
+        if self.begin == self.end:
+            # add the object itself
+            graph.add((region, RDF.type, faldo.ExactPosition))
+            graph.add((region, faldo.position, Literal(self.begin)))
+        else:
+            # add the object itself
+            graph.add( (region, RDF.type, faldo.Region) )
+
+            # add its properties, recursively if needed
+            graph.add( (region, faldo.begin, self.ns + "position/" + str(self.begin)) )
+            graph.add( (region, faldo.end, self.ns + "position/" + str(self.end)) )
+
+
+class Cell:
+    id: int
+    path_id: str
+    ns: URIRef
+    position_percent: float
+    inversion_percent: float
+    cell_region: List[Region] # [[1,4], [7,11]]
+
+    def __init__(self):
+        self.cell_region = []
+
+    def __str__(self):
+        return "cell" + self.path_id # cell1
+
+    def ns_term(self):
+        return self.ns + str(self)
+
+    def add_to_graph(self, graph: Graph, vg, faldo: Namespace) -> None:
+        cell = self.ns_term()
+        inner_ns = URIRef(self.path_id + "/")
```

Comment:
@6br @dimatr Do we want to name each cell with its path it is presenting, as given now.
Or do we want to count up the cell identifiers?

Reply:
I prefer the cell name is the primary path id and its subregion. But if the cell has more than one path, it would be matter.

Reply:
Our current model says a cell only has one path. So keep it as it is.

Reply:
With the discussion with Jerven, I suspect a cell can have one or more paths, and this would be the case.